packages/core-plugin-api/src/translation/useTranslationRef.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useTranslation } from 'react-i18next';
+
+import { TranslationOptions, TranslationRef } from './types';
+import { useApi } from '../apis';
+import { appTranslationApiRef } from '../apis/alpha';
+
+/** @alpha */
+export const useTranslationRef = <
+  Messages extends Record<keyof Messages, string>,
+>(
+  translationRef: TranslationRef<Messages>,
+) => {
+  const appTranslationApi = useApi(appTranslationApiRef);
+
+  appTranslationApi.addResourcesByRef(translationRef);
+
+  const { t } = useTranslation(translationRef.getId());
```

Comment:
Noting here that `react-i18next` does not deduplicate the context and we may need to contribute or work around that. A potential direct is to avoid using `useTranslation` here altogether and have it managed through the translation API instead. That'll have the benefit of being able to use translations in Utility API implementations too

Reply:
You means use `i18next.t` directly ?

Reply:
Yep, we'll need to explore either that or getting the context deduplicated. I'm definitely biased towards the former approach for the reason mentioned above.